api/controllers/AuthController.js

```diff
@@ -46,6 +46,10 @@ module.exports = {
   },
 
   callback: function(req, res) {
+    if (req.query.state !== req.session.state) {
+      console.log("Warning: A user was redirected to the reddit callback, but did not have a valid session state.");
+      return res.forbidden();
+    }
```

Comment:
Basically, now we only accept a callback from reddit if the login process started from our app. Before, a user could cause another user to become logged into the wrong account by getting them to click a link such as

    hq.porygon.co/auth/reddit/callback?state=anystring&code=valid_code_for_someone_elses_account

As far as I can tell, this isn't a huge security issue, because I'm not sure why an attacker would want to log a victim into someone else's account. (The attacker would have to have access to this other account in order to get a valid code for it.) Still, it's better just to avoid the possibility altogether -- people shouldn't get logged in unless they deliberately click the 'Log In' button.